src/mgwasm.c

```diff
@@ -27,6 +28,16 @@ int write_loop(const int sock) {
   if (poll == -1) {
     return -1;
   }
+  int result;
+  socklen_t result_len = sizeof(result);
+  if (getsockopt(sock, SOL_SOCKET, SO_ERROR, &result, &result_len) < 0) {
```

Comment:
Just to be sure: because connect is not synchronous in WASM, it is possible that it returns before the connection is done, so it can silently fail. Therefore before every write with `getsockopt` we check if there is any error (`SO_ERROR`). Right?

Reply:
Yep that's right!